### PR TITLE
feat(PreMiD): add buttons, presence/user icon

### PR DESCRIPTION
**Pull request by 7loud**
**Describe the changes in this pull request:**
* Added presence icon (`/store/presences/...`)
* Added user icon (`/users/...`)
* Added buttons

<details>
<summary> Proof showing the creation/modification is working as expected </summary>
<br>

image
Screenshot_120
Screenshot_118
</details>

### CODE REVIEW

**Review by Bas950 — DISMISSED** [review]

File: websites/P/PreMiD/presence.ts [hunk 1]
```diff
@@ -57,13 +58,23 @@ presence.on("UpdateData", async () => {
 			largeImageKey: "lg"
 		},
 		newLang = await presence.getSetting<string>("lang"),
-		time = await presence.getSetting<boolean>("time");
+		time = await presence.getSetting<boolean>("time"),
+		showButtons = await presence.getSetting<boolean>("showButtons");
```
Comment (EncryptedDev) [inline websites/P/PreMiD/presence.ts:62]: use Promise.all

File: websites/P/PreMiD/presence.ts [hunk 2]
```diff
@@ -57,13 +58,23 @@ presence.on("UpdateData", async () => {
...
 
 	if (oldLang !== newLang || !strings) {
 		oldLang = newLang;
 		strings = await getStrings();
 	}
 
+	if (showButtons) {
+		presenceData.buttons = [
+			{
+				label: (await strings).btnViewPage,
```
Comment (EncryptedDev) [inline websites/P/PreMiD/presence.ts:72]: You don't need to `await` if you have the Awaited type

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/P/PreMiD/presence.ts (hunk 2 above)
Comment (N0chteil) [reply websites/P/PreMiD/presence.ts:72]: So should I replace all `(await strings)` or just this one?

File: websites/P/PreMiD/presence.ts [hunk 3]
```diff
@@ -254,3 +274,19 @@ presence.on("UpdateData", async () => {
 	}
 	presence.setActivity(presenceData);
 });
+
+const shortenedURLs: Record<string, string> = {};
+async function getShortURL(url: string) {
```
Comment (Slowlife01) [inline websites/P/PreMiD/presence.ts:279]: I don't think this is necessary

Comment (N0chteil) [reply websites/P/PreMiD/presence.ts:279]: Discord pfp url?

Comment (Slowlife01) [reply websites/P/PreMiD/presence.ts:279]: Still no

**Review by EncryptedDev — APPROVED** [review]
lgtm

**Review by Slowlife01 — APPROVED** [review]

**Pull request merged**